### PR TITLE
Feature/webrdfs web request translator

### PR DESCRIPTION
**Pull request by vnazz**
Create a file with static methods to translate protobufs to the appropriate webRDFS response.  The created responses follow the webHDFS api.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -26,6 +26,7 @@ add_subdirectory(rice-datanode)
 add_subdirectory(native-filesystem)
 add_subdirectory(test)
 add_subdirectory(storage)
+add_subdirectory(webRequestTranslator)
```
Comment (LINKIWI) [inline CMakeLists.txt]: Can you follow the existing naming convention:

```
web-request-translator
```

File: webRequestTranslator/source/webRequestTranslator.cc [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+     * Converts the RDFS namenode create response into the appropriate webRDFS response.
+     */
+    std::string getNamenodeCreateResponse(hadoop::hdfs::DatanodeInfoProto &dataProto, std::string requestLink) {
+      std::string res = std::string("HTTP/1.1 307 TEMPORARY_REDIRECT\n");
+
+      std::string delimiter = "/webhfs/v1/";
+      std::string restOfRequest = requestLink.substr(requestLink.find(delimiter) + delimiter.length(), requestLink.length());
+
+      hadoop::hdfs::DatanodeIDProto id = dataProto.id();
+      res += "Location: http://";
```
Comment (LINKIWI) [inline webRequestTranslator/source/webRequestTranslator.cc]: Do we have a solution for choosing between `http` and `https?` Maybe define it as an env var and conditionally choose one here based on that? cc @danielbalkum

File: webRequestTranslator/source/webRequestTranslator.cc [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+      res += std::to_string(contentOfFile.length());
+      res += "\n\n";
+
+      return res;
+    }
+  /**
+   * Converts the RDFS namenode read response into the appropriate webRDFS response.
+   */
+  std::string getNamenodeReadResponse(hadoop::hdfs::DatanodeInfoProto &dataProto, std::string requestLink) {
+    std::string res = std::string("HTTP/1.1 307 TEMPORARY_REDIRECT\n");
```
Comment (LINKIWI) [inline webRequestTranslator/source/webRequestTranslator.cc]: `s/_/ /g`

to conform to HTTP standard, [reference](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/307)

File: webRequestTranslator/source/webRequestTranslator.cc [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+    res += "\n\n";
+    res += contentOfFile;
+
+    return res;
+  }
+  /**
+   * Converts the RDFS datanode mkdir response into the appropriate webRDFS response.
+   */
+  std::string getMkdirResponse(hadoop::hdfs::DatanodeInfoProto &dataProto, std::string requestLink) {
+    return "HTTP/1.1 200 OK\n{\"boolean\":true}";
```
Comment (LINKIWI) [inline webRequestTranslator/source/webRequestTranslator.cc]: Do we need to specify `Content-Type` for this? ditto for mv, delete

Comment (vnazz) [reply webRequestTranslator/source/webRequestTranslator.cc]: @heliumj

**Review by LINKIWI — APPROVED** [review]
lgtm

**Pull request merged**